### ISSUE TITLE
Invisible color tags are included in "Part Of" expression

### ISSUE THREAD

**Issue by antoinech2** [SkriptLang/Skript#3212]
### Description
When you use a colored test, i.e: `"&2Test"` or `"<#e81d13>Test"`, the invisible tags, like `&2` are taken into account when using the `part of %string%` expression. So the output can be blank caracters, see belox:

### Steps to Reproduce
`set {test} to colored "&2Test"`
`send part of {test} from character 1 to character 2`
EXPECTED Output: `Te` (in green)
FOUND Output: nothing (blank)
`send part of {test} from character 1 to character 4`
EXPECTED Output: `Test` (in green)
FOUND Output: `Te` (in green)

So it's shifted by the 2 invisible color caracters.
It's exactly the same with or without `colored` before the text

It's ever worse with HEX color tag:
`set {test} to colored "<#566353>Test"`
`send part of {test} from character 1 to character 14`
OUTPUT: blank
`send part of {test} from character 1 to character 15`
OUTPUT: `T` (with the right color)

So with Hex color it's shifted by 14 characters !

### Expected Behavior
The color tags shouldn't be taken into account in the Part Of expression

### Errors / Screenshots
![image](https://user-images.githubusercontent.com/42295406/88461685-e9766e80-cea5-11ea-8b01-0ae6eef9bb3e.png)
![image](https://user-images.githubusercontent.com/42295406/88461692-027f1f80-cea6-11ea-937d-2519aa82e6fe.png)

### Server Information
* **Server version/platform:** Minecraft 1.16.1 running Paper 1.16.1-R0.1-SNAPSHOT
* **Skript version:** 2.5-alpha6(rev1)

**Comment by ShaneBeee** [comment]
is this `part of` expression from Skript? or is it from an addon? 
I cant seem to find it in Skript itself.

**Comment by antoinech2** [comment]
> is this `part of` expression from Skript? or is it from an addon?
> I cant seem to find it in Skript itself.

Yes of course ! It is the `Subtext` expression (for for telling the bad name)
http://skriptlang.github.io/Skript/expressions.html#ExprSubstring

**Comment by Pikachu920** [comment]
it's an alias of the subtext expr
https://github.com/SkriptLang/Skript/blob/6524d69f1e8583f917ea18c9fe526ec6df66dbfc/src/main/java/ch/njol/skript/expressions/ExprSubstring.java#L49

**Comment by ShaneBeee** [comment]
Ah ok.
Im not sure if this would be classified as a bug or not, but you could make use of `uncoloured %string%`

**Comment by APickledWalrus** [comment]
This *might* be related to the fact that hex colors are prefixed with `§x`

**Comment by antoinech2** [comment]
> Ah ok.
> Im not sure if this would be classified as a bug or not, but you could make use of `uncoloured %string%`

Yes but then, I won't have any color in my message, so it will be more complicated to obtain what I want...

**Comment by antoinech2** [comment]
Note that it worked in earlier versions of Skript (or Minecraft idk)

**Comment by APickledWalrus** [comment]
> 
> 
> Note that it worked in earlier versions of Skript (or Minecraft idk)

Did it? Are you able to test what version it did ignore them?

**Comment by antoinech2** [comment]
> > Note that it worked in earlier versions of Skript (or Minecraft idk)
> 
> Did it? Are you able to test what version it did ignore them?

I'll test that tomorrow and tell you 😉 But I'm pretty sure it was working

**Comment by antoinech2** [comment]
Hi again !
Well, it seems that it always been like this (I was thinking that it worked as I expected on 1.12 but it wasn't...).
So I don't know if this is really a bug, but it's not a very "logical" behaviour, because when you ask for a part of a text, you want real characters, not invisible character.
Anyway, that's not a new behaviour

**Comment by APickledWalrus** [comment]
This could probably be improved then, but it could also break a lot of current scripts

**Comment by Govindass** [comment]
> 
> 
> This could probably be improved then, but it could also break a lot of current scripts

I agree on this, I happen to use this expression purely for getting the color codes

**Comment by sovdeeth** [comment]
I am in favor of keeping the existing behavior, as those characters are very much part of the string, despite being invisible. Ignoring them is less logical to me.
This may be impacted if we shift to component based text, however.
